code/game/objects/items/weapons/tools.dm

```diff
@@ -348,22 +348,31 @@
 			return
 		tank.armed = 1
 		user.visible_message("[user] begins heating the [O].", "You start to heat the [O].")
-		message_admins("[key_name_admin(user)] is attempting a welder bomb at ([loc.x],[loc.y],[loc.z]) - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[O.x];Y=[O.y];Z=[O.z]'>JMP</a>")
-		if(do_after(user,100))
-			if(tank.defuse)
-				user.visible_message("[user] melts some of the framework on the [O].", "You melt some of the framework.")
-				tank.defuse = 0
+		message_admins("[key_name_admin(user)] is potentialy arming a welder bomb at ([loc.x],[loc.y],[loc.z]) - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[O.x];Y=[O.y];Z=[O.z]'>JMP</a>")
+		switch(alert("Are you sure you want to do this? It is quite dangerous and could get you in trouble.", "Heat up fuel tank", "No", "Yes"))
+			if("Yes")
+				message_admins("[key_name_admin(user)] accepted the warning prompt and are welder bombing at ([loc.x],[loc.y],[loc.z]) - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[O.x];Y=[O.y];Z=[O.z]'>JMP</a>")
+				to_chat(user, span("alert", "Heating the fueltank..."))
+				if(do_after(user,200))
+					if(tank.defuse)
+						user.visible_message("[user] melts some of the framework on the [O].", "You melt some of the framework.")
+						tank.defuse = 0
+						tank.armed = 0
+						return
+					message_admins("[key_name_admin(user)] triggered a fueltank explosion.")
```

Comment:
refactor this and the next line to
`log_and_message_admins("triggered a fuel tank explosion", user)`